react/components/ExtraData.tsx

```diff
@@ -74,11 +74,23 @@ const ExtraData: React.VFC<Props> = ({
   onBillingAddressChange,
   onDocumentChange,
 }) => {
-  const { cardLastDigits, payment } = useOrderPayment()
+  const { cardLastDigits, payment, paymentSystems } = useOrderPayment()
   const intl = useIntl()
   const [userDocument, setDocument] = useState<Field>(initialDocument)
 
+  const documentIsRequired = useMemo(
+    () =>
+      paymentSystems.find(
+        paymentSystem => paymentSystem.id === payment.paymentSystem
+      )?.requiresDocument ?? false,
+    [payment.paymentSystem, paymentSystems]
+  )
+
   const validateDocument = () => {
+    if (!documentIsRequired) {
```

Comment:
you must check for their value here, else we could proceed with an invalid document even if it is optional